### ISSUE TITLE
$getAuth(), provider is undefined

### ISSUE THREAD

**Issue by tgensol** [firebase/angularfire#759]
### Version info

**Angular:** 1.5.6

**Firebase:**3.0.3

**AngularFire:** 2.0.1

**Other (e.g. Node, browser, operating system) (if applicable):** Ionic

### Test case

The doc says that $getAuth() returns :

> uid, provider (string identifying the provider), auth (the authentication token payload), and expires (expiration time in seconds since the Unix epoch)

But

`var authData = $scope.authObj.$getAuth();`
authData.uid is okay, but authData.provider and authData.auth are undefined.

### Steps to reproduce

```
var authData = $scope.authObj.$getAuth();
var provider = authData.provider
var uid = authData.uid
```
### Expected behavior

If the doc is correct, provider should be a string, and it is currently "undefined"

### Actual behavior

provider is "undefined".

**Comment by katowulf** [comment]
Thanks for this. It looks like the doc is just a bit out of date. Will use this as a bug to track. To some extent, the data returned depends on the auth method being used and the scopes you've assigned.

See [Get a user's profile](https://firebase.google.com/docs/auth/web/manage-users#get_a_users_profile). And [Get a user's provider-specific information](https://firebase.google.com/docs/auth/web/manage-users#get_a_users_provider-specific_profile_information).

So long story short:

```
var user = firebase.auth().currentUser;
if (user != null) {
  user.providerData.forEach(function (profile) {
    console.log("Sign-in provider: "+profile.providerId);
    console.log("  Provider-specific UID: "+profile.uid);
    console.log("  Name: "+profile.displayName);
    console.log("  Email: "+profile.email);
    console.log("  Photo URL: "+profile.photoURL);
  });
}
```

**Comment by jwngr** [comment]
Thanks for opening the issue! @abehaskins may be able to handle this one. Otherwise, someone in the community can take on this docs bug.

**Comment by jwngr** [comment]
This should be resolved with #763. The field is now called `providerId`, not `provider`.
